index.js

```diff
@@ -86,6 +86,9 @@ module.exports = {
     // the modern JSX runtime does not require React to be in scope
     'react/react-in-jsx-scope': 0,
 
+    // do not enforce a particular style of component definition
+    'react/function-component-definition': 0,
```

Comment:
See https://github.com/airbnb/javascript/issues/2505; I'm pretty sure the current `eslint-config-airbnb` config is a mistake. I'll remove this depending on the response.

Reply:
It's confirmed to be a mistake but they want to make changes to the underlying rule which will likely take awhile. I haven't seen examples of code that would violate this rule in practice, so I'm going to leave the rule off for now. Can always re-enable it in the next major.